crates/sncast/tests/e2e/declare_deploy.rs

```diff
@@ -0,0 +1,404 @@
+    let output = snapbox.assert().success();
+    let json = last_line_as_json(output.as_stdout());
+
+    assert_eq!(output.as_stderr(), "");
+    assert_eq!(json["command"], "declare-deploy");
+    assert!(json["class_hash"].as_str().is_some());
+    assert!(json["contract_address"].as_str().is_some());
+    assert!(json["declare_transaction_hash"].as_str().is_some());
+    assert!(json["deploy_transaction_hash"].as_str().is_some());
+}
+
+#[test_case("oz"; "oz_account")]
+#[test_case("argent"; "argent_account")]
+fn test_happy_case_strk(account: &str) {
+    let salt = account.to_owned() + "_strk";
+    let tempdir =
+        duplicate_contract_directory_with_salt(CONTRACTS_DIR.to_string() + "/map", "put", &salt);
+
+    let accounts_file = &get_accounts_path(ACCOUNT_FILE_PATH)[..];
+
+    let args = vec![
+        "--accounts-file",
+        accounts_file,
+        "--account",
+        account,
+        "--int-format",
+        "--json",
+        "declare-deploy",
+        "--url",
+        URL,
+        "--contract-name",
+        MAP_CONTRACT_NAME,
+        "--fee-token",
+        "strk",
+    ];
+
+    let snapbox = runner(&args).current_dir(tempdir.path());
+    let output = snapbox.assert().success();
+    let json = last_line_as_json(output.as_stdout());
+
+    assert_eq!(output.as_stderr(), "");
+    assert_eq!(json["command"], "declare-deploy");
+    assert!(json["class_hash"].as_str().is_some());
+    assert!(json["contract_address"].as_str().is_some());
+    assert!(json["declare_transaction_hash"].as_str().is_some());
+    assert!(json["deploy_transaction_hash"].as_str().is_some());
+}
+
+#[test]
+fn test_happy_case_human_readable() {
+    let tempdir = duplicate_contract_directory_with_salt(
+        CONTRACTS_DIR.to_string() + "/map",
+        "put",
+        "_human_readable",
+    );
+
+    let accounts_file = &get_accounts_path(ACCOUNT_FILE_PATH)[..];
+
+    let args = vec![
+        "--accounts-file",
+        accounts_file,
+        "--account",
+        "user17",
+        "--int-format",
+        "declare-deploy",
+        "--url",
+        URL,
+        "--contract-name",
+        MAP_CONTRACT_NAME,
+        "--fee-token",
+        "strk",
+    ];
+
+    let snapbox = runner(&args).current_dir(tempdir.path());
+    let output = snapbox.assert().success();
+
+    let expected = indoc!(
+        "
+        [..]
+        command: [..]
+        class_hash: [..]
+        contract_address: [..]
+        declare_transaction_hash: [..]
+        deploy_transaction_hash: [..]
+
+        To see declaration and deployment details, visit:
+        class: [..]
+        contract: [..]
+        declaration transaction: [..]
+        deployment transaction: [..]
+        "
+    );
+
+    assert_stdout_contains(output, expected);
+}
+
+#[test]
+#[ignore = "Expand the contract's code to more complex or wait for fix: https://github.com/xJonathanLEI/starknet-rs/issues/649#issue-2469861847"]
```

Comment:
This issue seems to be resolved. If this still cannot be addressed, let's create an issue and link TODO here.